pkg/storage/etcd/etcd_helper.go

```diff
@@ -558,6 +576,15 @@ func (h *etcdHelper) GuaranteedUpdate(ctx context.Context, key string, ptrToType
 			ttl = *newTTL
 		}
 
+		// Since update object may have a resourceVersion set, we need to clear it here.
+		if h.versioner != nil {
+			if err := h.versioner.UpdateObject(ret, meta.Expiration, 0); err != nil {
```

Comment:
I'm fine with punting on a self link fix. That'd only be a problem with objects available from multiple URLs, and we won't have many for 1.2.